app/controllers/archived_emails_controller.rb

```diff
@@ -0,0 +1,95 @@
+# frozen_string_literal: true
+
+class ArchivedEmailsController < ApplicationController
+  include ActionView::Helpers::TextHelper
+
+  before_action :authenticate_user!
+  before_action :load_resources
+  before_action :check_billing_access
+
+  def show
+    mail = Mail.new(@log_event.email_content)
+    prepare_email_data(mail)
+    render :show
+  rescue StandardError
```

Comment:
Is there a more specific error we could check? My guess is that in a few cases we may want to show the user an error message but otherwise the 500 page should be fine